src/java/org/apache/cassandra/index/sai/disk/v1/postings/MergePostingList.java

```diff
@@ -117,23 +114,27 @@ public long advance(long targetRowID) throws IOException
         // we can use it for the fast path
         long nextRowId = PostingList.END_OF_STREAM;
         PostingList nextPostingList = null;
-        var it = postingLists.iterator();
-        while (it.hasNext())
+        for (int i = 0; i < postingLists.size(); )
```

Comment:
We can prevent an array copy and use `i--` for all iterations of the for loop if we decrement through the list instead. This method is part of the `remove` method:

```java
/**
     * Private remove method that skips bounds checking and does not
     * return the value removed.
     */
    private void fastRemove(Object[] es, int i) {
        modCount++;
        final int newSize;
        if ((newSize = size - 1) > i)
            System.arraycopy(es, i + 1, es, i, newSize - i);
        es[size = newSize] = null;
    }
```